rp/src/exchange.rs

```diff
@@ -2,7 +2,9 @@ use std::{net::SocketAddr, path::PathBuf};
 
 use anyhow::Result;
 
+#[cfg(any(target_os = "linux", target_os = "freebsd"))]
```

Comment:
I would suggest to add a warning suppression here instead of a `cfg` option, as constants do not influence the compile result. Besides, maintaining this seems to be more cumbersome than an `#[allow(unused)]`.

Reply:
Lets do that in another PR